src/telemetry/extension.js

```diff
@@ -229,6 +231,19 @@ export async function getBadgeType() {
   }
 }
 
+export async function hasBadgeTextChanged(nextText, tabId = null) {
+  // If the nextText contains a digit, we know it's a price alert badge, which only affects
+  // the global badge text.
+  const returnGlobal = (/\d+/.test(nextText));
```

Comment:
Wouldn't `tabId` being null be a better signal that we want to check the global badge text?